### PR TITLE
CRUD functions for Inventories

### PR DESCRIPTION
**Pull request by ysoseriouz**
* Create nav-bar
* Inventories list with responsive table
* Create & Edit inventory pages
P/s: simple views without styling

### CODE REVIEW

File: app/views/inventories/destroy.html.erb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+<h1>Inventories#destroy</h1>
+<p>Find me in app/views/inventories/destroy.html.erb</p>
```
Comment (dptvn) [inline app/views/inventories/destroy.html.erb]: This page isn't neccessary

File: app/views/inventories/create.html.erb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+<h1>Inventories#create</h1>
+<p>Find me in app/views/inventories/create.html.erb</p>
```
Comment (dptvn) [inline app/views/inventories/create.html.erb]: This page isn't necessary.  You should delete this page

File: app/views/inventories/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        <th>Color</th>
+        <th>Status</th>
+        <th>Price</th>
+        <th colspan=2>Actions</th>
+      </tr>
+    </thead>
+
+    <tbody>
+      <% @inventories.each do |inventory| %>
+        <%# <% if friend.user == current_user %>
```
Comment (dptvn) [inline app/views/inventories/index.html.erb]: If this code doesn't use, should delete it

File: app/views/inventories/update.html.erb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+<h1>Inventories#update</h1>
+<p>Find me in app/views/inventories/update.html.erb</p>
```
Comment (dptvn) [inline app/views/inventories/update.html.erb]: This page isn't necessary.  You should delete this page

**Review by ducpt1012 — COMMENTED** [review]
Data and model could be not right.
image

**Review by ducpt1012 — COMMENTED** [review]
You could check this doc to validate at the model and return the better error.
https://guides.rubyonrails.org/active_record_validations.html

image

**Review by ducpt1012 — COMMENTED** [review]
You can use gem kaminari for paginating
https://github.com/kaminari/kaminari#installation.
We also have some gems for paginating at [here](https://rubygems.org/search?query=paginating), you could use any one of them.
image

File: app/helpers/inventories_helper.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  def manufactoring_year_range
+    1990..Time.zone.now.year
+  end
+
+  def inventory_status_options(inventory)
+    options_for_select(
+      Inventory.statuses.map { |k, v|
+        [k.titleize, Inventory.statuses.key(v)]
+      }, selected=inventory.status)
+  end
```
Comment (dptvn) [inline app/helpers/inventories_helper.rb:15]: There is a helper `select_tag` of rails support like this. 
https://apidock.com/rails/ActionView/Helpers/FormTagHelper/select_tag

File: app/models/model.rb [hunk 6]
```diff
@@ -1,4 +1,7 @@
 class Model < ApplicationRecord
   belongs_to :brand
-  has_many :inventories
+  has_many :inventories, dependent: :destroy
+
+  validates_associated :brand
+  validates :name, presence: true, uniqueness: { case_sensitive: false }
 end
```
Comment (dptvn) [inline app/models/model.rb]: Add a new line at end file.

File: app/views/inventories/_form.html.erb [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+        <% inventory.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div class="field form-group">
+    <%= form.label :model_id, class: "form-label" %>
+    <%= form.grouped_collection_select :model_id, Brand.order(:name), :models, :name, :id, :name, autofocus: true, class: "form-control"%>
```
Comment (dptvn) [inline app/views/inventories/_form.html.erb]: It seems class 'form-control' doesn't work for model, year, status
image

**Review by ducpt1012 — COMMENTED** [review]
IMO, `New inventory` should put inside `inventory` page, above inventories list. 
image

**Comment by ysoseriouz** [comment]
> IMO, `New inventory` should put inside `inventory` page, above inventories list.
> image

I think I can separate the navbar like this for better view, as I want the user can still see the same "New inventory" button when they come to edit page 🤔
Screen Shot 2021-07-13 at 09 17 56

File: app/views/inventories/_form.html.erb (hunk 7 above)
Comment (ysoseriouz) [reply app/views/inventories/_form.html.erb]: I've figured out how to fix it, you can check the effect again.

**Review by ducpt1012 — APPROVED** [review]

**Pull request merged**